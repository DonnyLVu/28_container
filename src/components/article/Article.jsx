import React from 'react';
import PropTypes from 'prop-types';

const Article = ({ title, author, description }) => {
  return (
    <>
      <a> {title} </a>
      <p>Author: {author}</p>
      <p>Description: {description}</p>
    </>
  );
};
Article.propTypes = {
  title: PropTypes.string.isRequired,
  author: PropTypes.string,
  description: PropTypes.string,
};
export default Article;
