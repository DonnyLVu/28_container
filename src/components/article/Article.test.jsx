import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Article from './Article';

describe('Article component', () => {
  afterEach(() => cleanup());
  it('renders Article', () => {
    const { asFragment } = render(<Article
      title="dummy data title"
      author="dummy data author"
      description="dummy data description" />);
    expect(asFragment()).toMatchSnapshot();
  });
});
