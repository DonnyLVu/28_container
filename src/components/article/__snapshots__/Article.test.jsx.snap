// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Article component renders Article 1`] = `
<DocumentFragment>
  <a>
     dummy data title 
  </a>
  <p>
    Author: dummy data author
  </p>
  <p>
    Description: dummy data description
  </p>
</DocumentFragment>
`;
